### ISSUE TITLE
[Bug] Helm installation fails when installing to a specific namespace

### ISSUE THREAD

**Issue by MinehubDE** [mariadb-operator/mariadb-operator#387]
**Documentation**
- [X] I acknowledge that I have read the relevant [documentation](https://github.com/mariadb-operator/mariadb-operator/tree/main/docs).

**Describe the bug**
When installing the Helm chart from the documentation, it fails when also providing a namespace while installing it. When installing without a specific namespace, it installs successfully in the default namespace.

**Expected behaviour**
Helm installs the resources in the provided namespace without failing.

**Steps to reproduce the bug**

1. Install the helm chart using the recommended installation instructions https://github.com/mariadb-operator/mariadb-operator?tab=readme-ov-file#recommended-installation
2. Add --namespace <my-namespace> to the command

**Debug information**
```bash
root@myhost ~ # helm install mariadb-operator mariadb-operator/mariadb-operator --namespace mariadb-operator --set metrics.enabled=true
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: [resource mapping not found for name: "mariadb-operator-cert-controller" namespace: "" from "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
ensure CRDs are installed first, resource mapping not found for name: "mariadb-operator" namespace: "" from "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
ensure CRDs are installed first, resource mapping not found for name: "mariadb-operator-webhook" namespace: "" from "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
ensure CRDs are installed first]
```
**Environment details**:
- Kubernetes version: v1.28.3
- Kubernetes distribution: MicroK8S
- mariadb-operator version: latest
- Install method: helm
- Install flavor: recommended

**Comment by jescarri** [comment]
@MinehubDE this error is because your cluster does not have the prometheus-operator CRDs installed, and you are setting 
````
--set metrics.enabled=true
`````

Try with:
````
--set metrics.enabled=false
````

Or install the prometheus operator first.

**Comment by MinehubDE** [comment]
When installing Helm charts, I'd expect the chart to install necessary subcharts in case they are needed.
Isn't there a possibility to add the needed command to the docs or to add extra arguments when installing the recommended version?

Apart from that i can confirm that the installation to a namspace works when leaving the metrics option aside.

**Comment by mmontes11** [comment]
> When installing Helm charts, I'd expect the chart to install necessary subcharts in case they are needed.
Isn't there a possibility to add the needed command to the docs or to add extra arguments when installing the recommended version?

The [recommended installation](https://github.com/mariadb-operator/mariadb-operator?tab=readme-ov-file#recommended-installation) has already links to the [prometheus operator](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)  and [cert-manager](https://cert-manager.io/docs/installation/) projects. They are considered de-facto instandards in Kubernetes for managing metrics and certificates respectively and adding specific installations on this repo feels a bit redundant TBH. 

If you are looking for testing/develop with `mariadb-operator`, our development guide suggest some handy way of installing them by using the make targets available in this project. See:
https://github.com/mariadb-operator/mariadb-operator/blob/main/docs/DEVELOPMENT.md#dependencies

Basically, clone this repo and run:
```bash
make install-prometheus
make install-cert-manager
```
